lib/kafka/consumer_group.rb

```diff
@@ -0,0 +1,170 @@
+require "set"
+require "kafka/protocol/member_assignment"
+
+module Kafka
+  class ConsumerGroup
+    attr_reader :assigned_partitions
+
+    def initialize(cluster:, logger:, group_id:)
+      @cluster = cluster
+      @logger = logger
+      @group_id = group_id
+      @session_timeout = 30
```

Comment:
Definitely – this is a use case specific tradeoff. Will make it configurable later and add some docs.